api/v1alpha1/config_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CephFsConfigSpec defines the desired CephFs configuration
+type CephFsConfigSpec struct {
+	SubVolumeGroup string `json:"subVolumeGroup,omitempty"`
+}
+
+// RbdConfigSpec defines the desired RBD configuration
+type RbdConfigSpec struct {
+	RadosNamespace string `json:"radosNamespace,omitempty"`
+}
+
+// NfsConfigSpec cdefines the desired NFS configuration
+type NfsConfigSpec struct {
+}
+
+// ConfigSpec defines the desired state of Config
+type ConfigSpec struct {
+	CephClusterRef corev1.LocalObjectReference `json:"cephClusterRef,omitempty"`
```

Comment:
@nb-ohad can you remove omitempty from this one, we need to have it set

Reply:
Omit empty is a serialization directive it does not indicate that the field is optional or not, it just mean that the filed will not be serialized on write if the value is empty.
 
For an optional field, we need an `+optional` comment above the field, so OpenAPI will treat it as optional (which we don't have right now for anything).

But regardless I can remove the omitempty if you feel it is necessary

Reply:
okay, it looks like i was not clear, i wanted to get it removed as its deadcode as we always set this value and its never going to be empty

Reply:
Yep, I can do that. 
I usually set omitempty for all fields of my APIs to lower the footprint when manually marshaling (for example in unit test code)

Reply:
done